clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java

```diff
@@ -74,7 +74,7 @@
         "their data.";
 
     public static final String MAX_MESSAGE_BYTES_CONFIG = "max.message.bytes";
-    public static final String MAX_MESSAGE_BYTES_DOC = "This is largest message size Kafka will allow to be " +
+    public static final String MAX_MESSAGE_BYTES_DOC = "This is largest record batch size Kafka will allow to be " +
```

Comment:
Yeah, the config name is annoying. I would document the current behaviour first and then have additional sentences for the older formats (and probably a brief reference to the unintuitive naming, which should come naturally after the older formats are explained).

Also, we should probably change `KafkaConfig.MessageMaxBytesDoc` to refer to this variable so that the docs remain consistent.